### PR TITLE
Make sure WindowsZip/LinuxTar remove correct dir when uninstall

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Make sure WindowsZip/LinuxTar remove correct dir when uninstall

### Issues Resolved
https://github.com/opensearch-project/opensearch-ci/issues/281
https://github.com/opensearch-project/opensearch-build/issues/3743

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peterzhuamazon** [comment]
Before:
```
rm: cannot remove 'C:\Users\Administrator\tmp76kgq8lv\local-test-cluster/opensearch-2.9.0/modules/transport-netty4/netty-buffer-4.1.94.Final.jar': Device or resource busy
rm: cannot remove 'C:\Users\Administrator\tmp76kgq8lv\local-test-cluster/opensearch-2.9.0/modules/transport-netty4/netty-codec-4.1.94.Final.jar': Device or resource busy
rm: cannot remove 'C:\Users\Administrator\tmp76kgq8lv\local-test-cluster/opensearch-2.9.0/modules/transport-netty4/netty-codec-http-4.1.94.Final.jar': Device or resource busy
rm: cannot remove 'C:\Users\Administrator\tmp76kgq8lv\local-test-cluster/opensearch-2.9.0/modules/transport-netty4/netty-common-4.1.94.Final.jar': Device or resource busy
rm: cannot remove 'C:\Users\Administrator\tmp76kgq8lv\local-test-cluster/opensearch-2.9.0/modules/transport-netty4/netty-handler-4.1.94.Final.jar': Device or resource busy
rm: cannot remove 'C:\Users\Administrator\tmp76kgq8lv\local-test-cluster/opensearch-2.9.0/modules/transport-netty4/netty-resolver-4.1.94.Final.jar': Device or resource busy
rm: cannot remove 'C:\Users\Administrator\tmp76kgq8lv\local-test-cluster/opensearch-2.9.0/modules/transport-netty4/netty-transport-4.1.94.Final.jar': Device or resource busy
rm: cannot remove 'C:\Users\Administrator\tmp76kgq8lv\local-test-cluster/opensearch-2.9.0/modules/transport-netty4/netty-transport-native-unix-common-4.1.94.Final.jar': Device
 or resource busy

......

2023-09-15 02:13:09 WARNING  Removing try count: 1/3 for C:\Users\Administrator\tmp76kgq8lv
2023-09-15 02:13:09 WARNING  Exception: [WinError 145] The directory is not empty: 'C:\\Users\\Administrator\\tmp76kgq8lv'
2023-09-15 02:13:09 WARNING  Removing try count: 2/3 for C:\Users\Administrator\tmp76kgq8lv
2023-09-15 02:13:09 WARNING  Exception: [WinError 145] The directory is not empty: 'C:\\Users\\Administrator\\tmp76kgq8lv'
2023-09-15 02:13:09 WARNING  Removing try count: 3/3 for C:\Users\Administrator\tmp76kgq8lv
2023-09-15 02:13:09 WARNING  Exception: [WinError 145] The directory is not empty: 'C:\\Users\\Administrator\\tmp76kgq8lv'
Traceback (most recent call last):
  File "C:\Users\Administrator\opensearch-build-peterzhuamazon\src\run_integ_test.py", line 36, in <module>

```

After:
```
2023-09-15 02:08:24 INFO     Sending SIGKILL to PID 13080
2023-09-15 02:08:24 INFO     Process killed with exit code None
2023-09-15 02:08:42 INFO     Recording local cluster logs for mlCommonsDashboards with test configuration as without-security at C:\Users\Administrator\opensearch-build-peterz
huamazon\test-results\ed8898ca2f2a4f77972b12e6479b9c75\integ-test\mlCommonsDashboards\without-security\local-cluster-logs
2023-09-15 02:08:42 INFO     Cleanup C:\Users\Administrator\tmp3zukb0i0\local-test-cluster\opensearch-dashboards-2.9.0 content after the test
2023-09-15 02:08:52 INFO     Sending SIGKILL to PID 14052
2023-09-15 02:08:52 INFO     Process killed with exit code None
2023-09-15 02:09:11 INFO     Recording local cluster logs for mlCommonsDashboards with test configuration as without-security at C:\Users\Administrator\opensearch-build-peterz
huamazon\test-results\ed8898ca2f2a4f77972b12e6479b9c75\integ-test\mlCommonsDashboards\without-security\local-cluster-logs
2023-09-15 02:09:11 INFO     Cleanup C:\Users\Administrator\tmp3zukb0i0\local-test-cluster\opensearch-2.9.0 content after the test
2023-09-15 02:09:11 INFO     Recording component test results for mlCommonsDashboards at C:\Users\Administrator\opensearch-build-peterzhuamazon\test-results\ed8898ca2f2a4f7797
2b12e6479b9c75\integ-test\mlCommonsDashboards\without-security
2023-09-15 02:09:11 INFO     Walking tree from C:\Users\Administrator\opensearch-build-peterzhuamazon\test-results\ed8898ca2f2a4f77972b12e6479b9c75\integ-test\mlCommonsDashboa
rds\without-security
2023-09-15 02:09:11 INFO     Removing C:\Users\Administrator\tmp3zukb0i0
2023-09-15 02:09:11 WARNING  Removing try count: 1/3 for C:\Users\Administrator\tmp3zukb0i0\mlCommonsDashboards\.git\objects\pack\pack-d9a227460d89739d54be0f92bdebcf87290503a7
.idx
2023-09-15 02:09:11 WARNING  Removing try count: 1/3 for C:\Users\Administrator\tmp3zukb0i0\mlCommonsDashboards\.git\objects\pack\pack-d9a227460d89739d54be0f92bdebcf87290503a7
.pack
2023-09-15 02:09:11 WARNING  Removing try count: 1/3 for C:\Users\Administrator\tmp3zukb0i0\mlCommonsDashboards\.git\objects\pack\pack-d9a227460d89739d54be0f92bdebcf87290503a7
.rev
2023-09-15 02:09:14 ERROR    | mlCommonsDashboards  | without-security     | FAIL  |

```

**Review by rishabh6788 — APPROVED** [review]

**Comment by rishabh6788** [comment]
Please fix the failing test @peterzhuamazon

File: src/test_workflow/integ_test/distribution_tar.py [hunk 1]
```diff
@@ -39,5 +39,5 @@ def start_cmd(self) -> str:
         return start_cmd_map[self.filename]
 
     def uninstall(self) -> None:
-        logging.info(f"Cleanup {self.work_dir}/* content after the test")
-        subprocess.check_call(f"rm -rf {self.work_dir}/*", shell=True)
+        logging.info(f"Cleanup {self.install_dir} content after the test")
+        subprocess.check_call(f"rm -rf {self.install_dir}", shell=True)
```
Comment (gaiksaya) [inline src/test_workflow/integ_test/distribution_tar.py:43]: Why not use path.join instead?

**Comment by peterzhuamazon** [comment]
> Please fix the failing test @peterzhuamazon

Fixed now.

File: src/test_workflow/integ_test/distribution_tar.py (hunk 1 above)
Comment (peterzhuamazon) [reply src/test_workflow/integ_test/distribution_tar.py:43]: I just fixed.

**Pull request merged**